label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRule data = 2;
+}
+
+message CreateLabelApplicationRule {
+  LabelApplicationRule data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  string description = 2;
+  MatchingCriteria matching_criteria = 3;
+  ApplicationCriteria application_criteria = 4;
+
+  message MatchingCriteria {
+    Type type = 1;
+    repeated Condition conditions = 2;
+
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      TYPE_SPAN = 1;
+      TYPE_TRACE = 2;
+    }
+
+    message Condition {
+      string name = 1;
+      Scope scope = 2;
+      string key = 3;
+      Operator operator = 4;
+      optional string value = 5;
+
+      enum Scope {
+        SCOPE_UNSPECIFIED = 0;
+        SCOPE_ATTRIBUTE = 1;
+        SCOPE_REQUEST = 2;
+        SCOPE_RESPONSE = 3;
+      }
+
+      enum Operator {
+        OPERATOR_UNSPECIFIED = 0;
+        OPERATOR_EXISTS = 1;
+        OPERATOR_EQUALS = 2;
+        OPERATOR_CONTAINS = 3;
+        OPERATOR_STARTS_WITH = 4;
```

Comment:
What's contains vs exists? Is contains operating on the value but exists only on the key with value being ignored? Operator names probably could use some clarification. Also, aren't contains/starts with/ends with just regex cases - do we need to support them as top level operators?

Reply:
You have captured the difference between exists and contains correctly. Let me add comments to provide clarity.

For string values will like to support contains, ends-with and starts-with as first class operators. This is similar to what we have java.lang.String.

Reply:
Is there a requirement somewhere for those operators? If not, I'd punt and can always add later. The difference in java is that convenience is a one-level thing - it's syntax sugar. Here, the addition of each operation spans many different parts of the system, so they're fairly expensive and we should be conservative about the ones we expose. The application of the label needs to know about it (implementation), the external API needs to support exposing it (serialization), the UI needs to support building it (presentation) etc.

Reply:
I don't think we have any specific requirement around operator. What do you think we need to support at minimum?

Reply:
I'd probably just start with exists/equals/regex

Reply:
Done